src/Microsoft.AspNet.Authentication.OpenIdConnect/OpenIdConnectAuthenticationHandler.cs

```diff
@@ -217,6 +218,47 @@ public OpenIdConnectAuthenticationHandler(HttpClient backchannel)
 
             message.State = Options.StateDataFormat.Protect(properties);
 
+            if (Options.UsePostAuthorizationRequests)
+            {
+                using (var buffer = new MemoryStream())
+                using (var writer = new StreamWriter(buffer))
+                {
+                    writer.WriteLine("<!doctype html>");
+                    writer.WriteLine("<html>");
+                    writer.WriteLine("<body>");
+
+                    writer.WriteLine("<form name='form' method='post' action='" + Options.HtmlEncoder.HtmlEncode(message.IssuerAddress) + "'>");
+
+                    foreach (var parameter in message.Parameters)
+                    {
+                        var name = Options.HtmlEncoder.HtmlEncode(parameter.Key);
+                        var value = Options.HtmlEncoder.HtmlEncode(parameter.Value);
+
+                        writer.WriteLine("<input type='hidden' name='" + name + "' value='" + value + "' />");
+                    }
+
+                    writer.WriteLine("<noscript>Click here to finish the authorization process: <input type='submit' /></noscript>");
+                    writer.WriteLine("</form>");
+                    writer.WriteLine("<script>document.form.submit();</script>");
+                    writer.WriteLine("</body>");
+                    writer.WriteLine("</html>");
+                    writer.Flush();
+
+                    Response.ContentLength = buffer.Length;
+                    Response.ContentType = "text/html;charset=UTF-8";
+
+                    // Emit Cache-Control=no-cache to prevent client caching.
+                    Response.Headers.Set("Cache-Control", "no-cache");
```

Comment:
We could generalize this practice and also emit these headers for GET requests, but AFAIK, browsers are not supposed to cache 302 responses so I only applied it for POST requests.

/cc @Tratcher